src/bbs-plus-presentation.js

```diff
@@ -0,0 +1,69 @@
+import { BBSPlusPublicKeyG2 } from '@docknetwork/crypto-wasm-ts';
+import b58 from 'bs58';
+import {
+  PresentationBuilder,
+  Credential,
+} from '@docknetwork/crypto-wasm-ts/lib/anonymous-credentials';
+import { ensureArray } from './utils/type-helpers';
+
+import Bls12381BBSSignatureDock2022 from './utils/vc/crypto/Bls12381BBSSignatureDock2022';
+import CustomLinkedDataSignature from './utils/vc/crypto/custom-linkeddatasignature';
+
+export default class BbsPlusPresentation {
+  /**
+   * Create a new BbsPlusPresentation instance.
+   * @constructor
+   */
+  constructor() {
+    this.presBuilder = new PresentationBuilder();
+  }
+
+  /**
+   * Species attributes to be revealed in the credential
+   * @param {number} credentialIndex
+   * @param {Array.<string>} attributes
+   */
+  addAttributeToReveal(credentialIndex, attributes = []) {
+    ensureArray(attributes);
+    this.presBuilder.markAttributesRevealed(credentialIndex, new Set(attributes));
+  }
+
+  /**
+   * Create bbs presentation
+   * @param options
+   * @returns {object}
+   */
+  createPresentation(options = {}) {
+    const { nonce } = options;
+    if (nonce && typeof nonce === 'string') {
+      this.presBuilder.nonce = b58.decode(nonce);
```

Comment:
this doesnt make much sense to enforce nonce to always be b58 encoded, also doesnt account for if nonce is not string (what if user wants to pass uint8array directly?)

Reply:
use the `stringToU8a` method from @polkadot/util (theres examples of this in the code if you search for it, quite simple method)